sys/net/link_layer/iQueue_MAC/include/mac_driver.h

```diff
@@ -0,0 +1,189 @@
+     */
+    mac_return_t (* off)(void);
+
+    /**
+     * @brief Indicates whether the driver is active (on) or not (off).
+     * @return A @c true @c value if the MAC driver is currently active,
+     *         a @c false @c value otherwise.
+     */
+    bool (* is_active)(void);
+
+    /**
+     * @brief Send a packet using this MAC/RDC protocol.
+     * @details Call this function when you want to enqueue
+     *          a packet for sending. Note that the protocol has
+     *          full control as to when to actually send that packet.
+     *
+     * @param[in] buf Buffer containing the packet to transmit
+     *                in a raw format (untyped pointer to data).
+     * @param[in] len Length, in bytes, of the raw data (packet)
+     *                to transmit from @c buf @c.
+     * @param[out] tx_status_callback Callback function that the MAC/RDC
+     *                                protocol invokes to notify the user
+     *                                about the on-going status this packet's
+     *                                transmission; since MAC/RDC protocols
+     *                                normally delay packets transmission,
+     *                                this callback function is the way
+     *                                information about TX unfolding can
+     *                                be eventually given back to user code.
+     *                                If you don't care about TX outcome,
+     *                                you can pass NULL as a value to this
+     *                                parameter.
+     * @param[out] tx_callback_param Arbitrary parameter to be passed to
+     *                               the @c tx_callback_function @c when
+     *                               this packet is concerned; can be NULL.
+     *
+     * @return a status value indicating how the operation proceeded
+     *         in the instant. For information about long-term status
+     *         of the transmission, use the @c tx_status_callback @c
+     *         parameter.
+     */
+    mac_return_t (* send)(void *buf,
+                          unsigned int len,
+                          mac_tx_status_callback_t tx_status_callback,
+                          void *tx_callback_param);
+
+    /**
+     * @brief Receive a packet from radio transceiver
+     *        using this MAC/RDC protocol.
+     *
+     * @see mac_rx_packet_notif_t
+     */
+    void (* receive)(void *buf,
+                     unsigned int len,
+                     uint8_t rssi,
+                     uint8_t lqi,
+                     bool crc_ok);
+
+    /**
+     * @brief Define incoming packet callback function.
+     * @details Define the callback function that the MAC/RDC protocol
+     *          invokes when it has received a packet from network interface.
+     */
+    void (* set_input_func)(mac_rx_packet_notif_t rx_func);
+
+} mac_driver_t;
```

Comment:
Please also rename the functions and structs here, e.g. `typedef struct {...} iqueue_mac_driver_t`.